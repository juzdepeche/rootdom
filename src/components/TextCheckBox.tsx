import React, { useState } from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';
import CheckBox from '@react-native-community/checkbox';

type TextCheckBoxProps = {
	text: string;
	selected: boolean;
	onFactionToggle: (factionName: string, isSelected: boolean) => void;
};

export const TextCheckBox = ({
	text,
	selected,
	onFactionToggle
}: TextCheckBoxProps) => {
	const [isSelected, setIsSelected] = useState(selected);

	const onToggle = () => {
		onFactionToggle(text, !isSelected);
		setIsSelected(!isSelected);
	};

	return (
		<TouchableOpacity onPress={() => onToggle()} style={styles.item}>
			<CheckBox value={isSelected} onChange={() => onToggle()} />
			<Text
				style={[
					{
						...styles.checkBoxTxt,
						color: isSelected ? '#364f6b' : 'gray',
						fontWeight: isSelected ? 'bold' : 'normal',
						fontSize: 20
					}
				]}
				onPress={() => onToggle()}
			>
				{text}
			</Text>
		</TouchableOpacity>
	);
};

export default TextCheckBox;

const styles = StyleSheet.create({
	item: {
		width: '100%',
		backgroundColor: '#f6f6f6',
		borderRadius: 20,
		padding: 10,
		marginBottom: 10,
		flexDirection: 'row'
	},
	checkBoxTxt: {
		marginLeft: 20
	}
});
