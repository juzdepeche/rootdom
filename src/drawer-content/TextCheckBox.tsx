import { CheckBox } from 'native-base';
import React, { useState } from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';

type TextCheckBoxProps = {
	text: string;
	onFactionToggle: (factionName: string, isSelected: boolean) => void;
};

export const TextCheckBox = ({ text, onFactionToggle }: TextCheckBoxProps) => {
	const [isSelected, setIsSelected] = useState(true);

	const onToggle = () => {
		onFactionToggle(text, !isSelected);
		setIsSelected(!isSelected);
	};

	return (
		<TouchableOpacity onPress={() => onToggle()} style={styles.item}>
			<CheckBox
				checked={isSelected}
				color="#364f6b"
				onPress={() => onToggle()}
			/>
			<Text
				style={[
					{
						...styles.checkBoxTxt,
						color: isSelected ? '#364f6b' : 'gray',
						fontWeight: isSelected ? 'bold' : 'normal',
						fontSize: 20
					}
				]}
				onPress={() => onToggle()}
			>
				{text}
			</Text>
		</TouchableOpacity>
	);
};

const styles = StyleSheet.create({
	item: {
		width: '100%',
		backgroundColor: '#f6f6f6',
		borderRadius: 20,
		padding: 10,
		marginBottom: 10,
		flexDirection: 'row'
	},
	checkBoxTxt: {
		marginLeft: 20
	}
});
