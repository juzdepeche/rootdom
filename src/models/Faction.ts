import images from '../../assets/images';

export class Faction {
	public name: string;
	public value: number;
	public color: string;
	public image;

	constructor(name: string, value: number, color: string, image?) {
		this.name = name;
		this.value = value;
		this.color = color;
		this.image = image;
	}
}

export const Factions = [
	new Faction('Marquise de Cat', 10, '#FD5200', images.cat),
	new Faction('Underground Duchy', 8, '#D7A872', images.taupe),
	new Faction('Eyrie Dynasties', 7, '#4774B7', images.bird),
	new Faction('Vagabond', 5, '#FCFEFD', images.vagabond),
	new Faction('Riverfolk Company', 5, '#61DDFB', images.otter),
	new Faction('Woodland Alliance', 3, '#59BA53', images.mouse),
	new Faction('Corvid Consipiracy', 3, '#290D7E', images.raven),
	new Faction('Lizard Cult', 2, '#f1f435', images.lizard)
];

export const GetViableGameSum = (playerCount:number) => {
	switch (playerCount) {
		case 2:
			return 17;
		case 3:
			return 18;
		case 4:
			return 21;
		case 5:
			return 25;
		case 6:
			return 28;
		default:
			return 0;
	}
}

export const GetBigFactionsSum = (factions:Faction[], playerCount:number) => {
	let sum = 0;
	factions.sort((a, b) => a.value > b.value ? -1 : 1)
	for (let i = 0; i < playerCount; i++) {
		sum += factions[i].value;
	}
	return sum;
}