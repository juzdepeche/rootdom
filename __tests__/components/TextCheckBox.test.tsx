import React from 'react';
import renderer from 'react-test-renderer';
import { Factions } from '../../src/models/Faction';
import TextCheckBox from '../../src/components/TextCheckBox';

describe('<DrawerContent />', () => {
	it('renders correctly', () => {
		const factionName = Factions[0].name;
		const onFactionToggle = () => true;
		const textCheckBox = renderer
			.create(
				<TextCheckBox
					key={factionName}
					text={factionName}
					onFactionToggle={onFactionToggle}
				></TextCheckBox>
			)
			.toJSON();
		expect(textCheckBox).toMatchSnapshot();
	});
});
