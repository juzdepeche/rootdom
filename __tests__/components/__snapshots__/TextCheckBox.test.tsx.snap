// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerContent /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#f6f6f6",
      "borderRadius": 20,
      "flexDirection": "row",
      "marginBottom": 10,
      "opacity": 1,
      "padding": 10,
      "width": "100%",
    }
  }
>
  <View
    pointerEvents="auto"
  >
    <RNCCheckbox
      forwardedRef={null}
      onChange={[Function]}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 32,
            "width": 32,
          },
          undefined,
        ]
      }
      value={true}
    />
  </View>
  <Text
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "#364f6b",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginLeft": 20,
        },
      ]
    }
  >
    Marquise de Cat
  </Text>
</View>
`;
